### ISSUE TITLE
can't make --sound work

### ISSUE THREAD

**Issue by rsyqvthv** [i2van/hourglass#16]
I am using portable version.. 

I download zip, unpack to the folder `hourglass`, followed by run `ngen-Hourglass.bat install`.
then I put a wav file `alarm.wav` in the `hourglass`, 

I tested using this command to start a 1 minute timer: `Hourglass.exe --sound alarm 1`, it's no sound when timer expired.

I've tried make a `Sounds` folder, and put alarm.wav in, same result.. 
and also tested mp3, aac, m4a, also didn't work.

what am I doing wrong?

**Comment by ivan-ivon** [comment]
Please run

`ngen-Hourglass.bat uninstall`

It should work after that. I'll fix it in next version.

**Comment by i2van** [comment]
Please run

`ngen-Hourglass.bat uninstall`

It should work after that. I'll fix it in next version.

**Comment by i2van** [comment]
@rsyqvthv You can download fix [here](https://github.com/i2van/hourglass/actions/runs/10918215205)

**Comment by i2van** [comment]
@rsyqvthv Have you tried a [build](https://github.com/i2van/hourglass/actions/runs/10918215205)?

**Comment by i2van** [comment]
Have you exited from Hourglass before running fixed version? If not old one will be proceeding your command.

**Comment by i2van** [comment]
Sound files should be copied to the corresponding folders **before** running any Hourglass. Otherwise error `Invalid value "<SOUND>" for switch "--sound"` will popup.

**Comment by i2van** [comment]
@rsyqvthv Please try latest [build](https://github.com/i2van/hourglass/actions). Works for me.

Sound files should be copied to the corresponding folders before running any Hourglass. Otherwise error Invalid value "<SOUND>" for switch "--sound" will popup.

**Comment by i2van** [comment]
@rsyqvthv Could you please try [this build](https://github.com/i2van/hourglass/actions/runs/10970918843)?

I added message boxes before playing custom sound and in case of error. Please copy content of message boxes using `Ctrl`+`C` and insert them in reply.

**Comment by i2van** [comment]
@rsyqvthv Was there a sound?

What about `.wav` instead of `.mp3`?

**Comment by i2van** [comment]
@rsyqvthv Could you please check your WMP version? It can be found at `C:\Program Files (x86)\Windows Media Player`

![image](https://github.com/user-attachments/assets/890acb5d-a0ed-464a-9fe5-a833f8abe2a5)

**Comment by i2van** [comment]
@rsyqvthv Could you do the [check whether WMP is installed](https://stackoverflow.com/a/4035587)?

> any possible hourglass can play mp3/wav sound without install the wmplayer?

I don't think so.

**Comment by i2van** [comment]
Please try [this build's](https://github.com/i2van/hourglass/actions/runs/10976224471) Portable Hourglass.

**Comment by i2van** [comment]
@rsyqvthv I hope I'll release prod version next week. That one you have it's just for proof of concept and not ready for production.

**Comment by i2van** [comment]
@rsyqvthv Could you please test [this build](https://github.com/i2van/hourglass/actions/runs/10984108534). You have to copy the files from `Hourglass.NAudio.zip` next to `Hourglass.exe`. Without them the default player will be used.

**Comment by i2van** [comment]
NAudio supports only the following:

- wav
- mp3
- aiff

**Comment by i2van** [comment]
Fixed in [1.15.38](https://github.com/i2van/hourglass/releases/tag/1.15.38)

**Comment by badgersaurus** [comment]
Hi, I'm having the same issue again. Have used the NAudio file and tried both a portable and local installation and a few different MP3 and WAV files. No luck.

**Comment by i2van** [comment]
@badgersaurus Have it been working before?

**Comment by badgersaurus** [comment]
@i2van No, only had this installed a few days. It shows up in the Sound menu and can be selected fine, just don't hear anything when the timer finishes. Regular beeps still work though.

**Comment by i2van** [comment]
@badgersaurus Have you extracted `NVorbis` files next to the `Hourglass.exe`? You have to restart Hourglass afterwards.

![Image](https://github.com/user-attachments/assets/89e8c685-522a-4caf-b7e7-0fd974119e62)

**Comment by badgersaurus** [comment]
@i2van Yes:

![Image](https://github.com/user-attachments/assets/6231bc12-fb32-4b43-8619-0c9be90bcb14)

Though I don't have a user.config (I guess probably because this is a local install and not a portable one ?)

**Comment by i2van** [comment]
@badgersaurus Could you please attach your sound file?

**Comment by badgersaurus** [comment]
@i2van https://f.uguu.se/kufGtiPV.wav Here it is. I've also tried with several .mp3 music files I have (copyrighted so probably shouldn't share here) and had no luck.

**Comment by i2van** [comment]
@badgersaurus Could you please use [this build](https://github.com/i2van/hourglass/actions/runs/14579394044)? It should crash if sound could not be played.

Error logs can be found [here](https://github.com/i2van/hourglass/blob/main/FAQ.md#what-should-i-do-if-the-hourglass-does-not-start).

**Comment by badgersaurus** [comment]
@i2van It doesn't crash, just ends silently like before. Tried the same file + a song again.

**Comment by i2van** [comment]
@badgersaurus Looks like I have to create NAudio app for playing sounds...

**Comment by i2van** [comment]
@badgersaurus Could you please play your files with [NAudio](https://github.com/naudio/NAudio) demo (attached)? .NET 6.0 is required. I built it locally ([virustotal report](https://www.virustotal.com/gui/file/0a009c00cc0dea427f11068ff7b001818de96f07abe4253c7aaa9efea2fdf802?nocache=1))

![Image](https://github.com/user-attachments/assets/edbf3518-a959-492d-9d15-d7aa01e1e072)

[NAudio-net6.0-windows.zip](https://github.com/user-attachments/files/19839916/NAudio-net6.0-windows.zip)

**Comment by badgersaurus** [comment]
@i2van Seems to work, I can hear it fine through there.

**Comment by i2van** [comment]
So,

1. Sound is selected in `Sounds` menu.
2. When timer completes sound is not played.
3. Sound can be played by demo app.

Could you please try different output drivers? Which one doesn't work?

![Image](https://github.com/user-attachments/assets/d73388d3-1e71-48e8-904a-74cea1a72ded)

**Comment by badgersaurus** [comment]
@i2van WasapiOut, WaveOut and Directsound all work. AsioOut just displays an error that it 'is unavailable on this system. '

**Comment by i2van** [comment]
@badgersaurus What is the Windows version? Please run `ver` in console.

**Comment by i2van** [comment]
@badgersaurus Please try this [NAudioHourglassPack](https://github.com/i2van/hourglass/actions/runs/14590027546/artifacts/2984145340). I set output device to `0` instead of default one.

**Comment by badgersaurus** [comment]
I'm on 64-bit, Version 10.0.26100.3775

The new Naudio doesn't seem to help unfortunately.

**Comment by i2van** [comment]
Please try [this build](https://github.com/i2van/hourglass/actions/runs/14628088927). Both Hourglass and NAudioHourglassPack.
